pkg/ccl/changefeedccl/cdctest/validator_test.go

```diff
@@ -95,6 +104,63 @@ func TestOrderValidator(t *testing.T) {
 	})
 }
 
+func TestBeforeAfterValidatorForGeometry(t *testing.T) {
+	defer leaktest.AfterTest(t)()
+	ctx := context.Background()
+	s, sqlDBRaw, _ := serverutils.StartServer(t, base.TestServerArgs{UseDatabase: "d"})
+	defer s.Stopper().Stop(ctx)
+	sqlDB := sqlutils.MakeSQLRunner(sqlDBRaw)
+	tsRaw := make([]string, 1)
+
+	sqlDB.Exec(t, `CREATE DATABASE d`)
+	sqlDB.Exec(t, `CREATE TABLE foo (k INT PRIMARY KEY,  geom geometry(point))`)
+	sqlDB.QueryRow(t, `INSERT INTO foo VALUES(1, 'point(1 2)') RETURNING cluster_logical_timestamp()`).Scan(&tsRaw[0])
+
+	ts := make([]hlc.Timestamp, len(tsRaw))
+	for i := range tsRaw {
+		var err error
+		ts[i], err = hlc.ParseHLC(tsRaw[i])
+		if err != nil {
+			t.Fatal(err)
+		}
+	}
+	v, err := NewBeforeAfterValidator(sqlDBRaw, `foo`)
+	require.NoError(t, err)
+	assertValidatorFailures(t, v)
+	noteRow(t, v, `p`, `[1]`, `{"after": {"k":1, "geom":{"coordinates": [1,2], "type": "Point"}}}`, ts[0])
+}
+
+func TestExtractFieldFromJSON(t *testing.T) {
+	defer leaktest.AfterTest(t)()
+	skip.WithIssue(t, 134904)
```

Comment:
Note that this test is skipped with https://github.com/cockroachdb/cockroach/issues/134904.